mkosi/config.py

```diff
@@ -462,8 +464,12 @@ def parse_path(value: str,
             path = INVOKING_USER.home() / path.relative_to("~")
         path = path.expanduser()
 
-    if required and not path.exists():
-        die(f"{value} does not exist")
+    if not path.exists():
+        if required:
+            die(f"{value} does not exist")
+        # If this is could be either a path or a string, and it doesn't exist, assume it's a string and return it.
+        if pathstring:
+            return Path(original)
```

Comment:
The name `pathstring` for this option seems a bit obscure, mostly because of our type `PathString`. Maybe a name like `assume_exists`? That being said the comment is a bit unclear. We don't need to assume something is a string, we know the value we have coming in and even use it as `original`. 

But looking at the only two users, I'm not sure I understand why this is needed. The users are:
```python
parse=config_make_path_parser(secret=True, required=False, pathstring=True),
```
but `secret=True` will not touch `path` if it doesn't exist, `absolute` is false by default and `resolve`, which is true by default, should be a noop for a non-existing path, so `required=False` should already suffice to skip this. The only difference then, is that this returns the path without expanding environment variables or the username, after expanding them and seeing that it doesn't exist. If it's about that a name a name for the option like `unexpand_vars_and_user_when_nonexistent` (or anything that's less of a mouthful)  might be better. 

Or just an option `passthrough` doing just that: returning `Path(value)` at the earliest possible moment.

Reply:
I've renamed to assume_exists - for the rest, this was necessary to stop all the type checking malarky from complaining at me. If you want it restructured in some way please feel free to push a change to the branch, after ensuring the CI is happy with it

Reply:
As far as I understand this piece, I see two reasonable things here
```python
def parse_path(value: str,
               *,
               required: bool = True,
               resolve: bool = True,
               expanduser: bool = True,
               expandvars: bool = True,
               secret: bool = False,
               absolute: bool = False,
               pass_through: bool = False,
               constants: Sequence[str] = ()) -> Path:
    if value in constants:
        return Path(value)

    if pass_through:
        return Path(value)

    # rest omitted
```
and
```python
def parse_path(value: str,
               *,
               required: bool = True,
               resolve: bool = True,
               expanduser: bool = True,
               expandvars: bool = True,
               secret: bool = False,
               absolute: bool = False,
               assume_exists: bool = False,
               constants: Sequence[str] = ()) -> Path:
    if value in constants:
        return Path(value)

    original = value

    if expandvars:
        value = os.path.expandvars(value)

    path = Path(value)

    if expanduser:
        if path.is_relative_to("~") and not INVOKING_USER.is_running_user():
            path = INVOKING_USER.home() / path.relative_to("~")
        path = path.expanduser()

    if assume_exists:
        return path
    # rest omitted
```

Depending on whether expanding environment variables or `~` is meant to be possible or not. Either type checked for me on a local checkout of your branch.

Reply:
please pick your favorite and just push it